libs/platform/table/components/table-view-settings-dialog/filtering/filters.component.ts

```diff
@@ -83,141 +62,152 @@ export enum ACTIVE_STEP {
-    filterBy: CollectionFilter[];
+    /** Emits the final result of filter changes */
+    filterChange = output<FiltersDialogResultData>();
 
-    /** Table columns. Used to retrieve a column key */
-    columns: TableColumn[];
+    /** Emits whether the reset button should be enabled */
+    resetAvailabilityChange = output<boolean>();
 
-    /** Declared filters options */
-    viewSettingsFilters: TableViewSettingsFilterComponent[];
+    /** Signal to hold currently applied filter rules */
+    filterBy = signal<CollectionFilter[]>([]);
 
-    /** Currently selected step */
-    activeStep: ACTIVE_STEP = ACTIVE_STEP.SELECT_FILTER;
+    /** Signal for table columns, used to retrieve the column key */
+    columns = signal<TableColumn[]>([]);
 
-    /** Currently selected filter to drill down */
-    activeFilter: TableViewSettingsFilterComponent | null = null;
+    /** Signal for declared filter options */
+    viewSettingsFilters = signal<TableViewSettingsFilterComponent[]>([]);
 
-    /** Table column key associated with the currently selected filter  */
-    activeFilterColumnKey: Nullable<string> = null;
+    /** Signal to track the currently selected filter step */
+    activeStep = signal<ACTIVE_STEP>(ACTIVE_STEP.SELECT_FILTER);
 
-    /** Indicates when reset command is available */
-    readonly isResetAvailable$ = signal(false);
+    /** Signal to track the currently selected filter */
+    activeFilter = signal<Nullable<TableViewSettingsFilterComponent>>(null);
 
-    /** Current step component to render */
-    activeFilterStepView: FiltersViewStep;
+    /** Signal to track the column key associated with the active filter */
+    activeFilterColumnKey = signal<Nullable<string>>(null);
 
-    /** @hidden */
-    private _initialFilters: CollectionFilter[] = [];
+    /** Signal for the currently active filter step view */
+    activeFilterStepView = signal<Nullable<FiltersViewStep>>(null);
 
-    /** @hidden */
-    constructor(
-        public readonly dialogRef: DialogRef<FiltersDialogData>,
-        private readonly _cd: ChangeDetectorRef,
-        private readonly _table: Table
-    ) {
-        const dialogData = this.dialogRef.data;
-        this.filterBy = [...dialogData.filterBy];
-        this.viewSettingsFilters = dialogData.viewSettingsFilters;
-        this.columns = dialogData.columns;
+    /** Reference to the available steps */
+    readonly ACTIVE_STEP = ACTIVE_STEP;
+
+    /** hidden */
+    constructor(private readonly _cd: ChangeDetectorRef) {
+        effect(
+            () => {
+                const filteringData = this.filteringData();
+                if (filteringData) {
+                    this.filterBy.set([...filteringData.filterBy]);
+                    this.viewSettingsFilters.set(filteringData.viewSettingsFilters);
+                    this.columns.set(filteringData.columns);
+                }
+            },
+            { allowSignalWrites: true }
+        );
```

Comment:
Angular strongly suggests against this approach. Have you tried achieving this with computed signals?